### ISSUE TITLE
Function parameter class types not properly colored

### ISSUE THREAD

**Issue by philraj** [atom/language-php#276]
### Description
Function parameter types which are using class names are not properly highlighted in yellow. Note `ClassNameA` and `ClassNameB` below:

![screen shot 2017-09-19 at 3 49 21 pm](https://user-images.githubusercontent.com/16470658/30612366-4951744a-9d52-11e7-9a62-e5018981295a.png)

### Versions
Atom v1.20.1

**Comment by 50Wliu** [comment]
This is generally consistent with how types are tokenized in other Atom languages.  The class name itself is tokenized as `entity.name.type.class` and all references to it are `storage.type`.

/cc @Ingramz for thoughts

**Comment by brysem** [comment]
I am experiencing the same issue on multiple setups of Atom on different MacBooks, including fresh installs. It's rather annoying as other packages experience issues such as specific features (jumping to definition) no longer work.

This was worked approximately two weeks ago. I assume there was an automatic update that created this issue.

Is there any debug information I can provide to help solve this problem?

The issue exists with namespaces at the top of PHP files as well. The whole class name used to be the same color. The last part is now only colored.
![screen shot 2017-09-20 at 08 51 06](https://user-images.githubusercontent.com/2829285/30630420-e1617daa-9de0-11e7-9f85-6aa464c3c823.png)

**Comment by 50Wliu** [comment]
Again, that change is intentional.  `App` and `Models` are tokenized as `support.other.namespace`, `\` is `punctuation.separator.inheritance`, and `Contact` is `support.class`.  This is to allow syntax themes finer-grained control over how to highlight the PHP file.  For example, this is how my current syntax theme highlights your example:
![php-different-syntax-themes](https://user-images.githubusercontent.com/2766036/30635564-13531e2e-9df3-11e7-9446-4f0455389603.png)

**Comment by Ingramz** [comment]
@brysem could you please provide an example of the packages that have stopped working properly because of language-php changes (and maybe also some instructions how to reproduce the issue you are experiencing).

As for the rest of the issue(s), I am still trying to figure out which parts should be changed by investigating similar usage elsewhere.

**Comment by philraj** [comment]
I'm in support of allowing the separate tokenization of these different cases, but I think it feels weird to have this much inconsistency in the coloring by default:

![screen shot 2017-09-21 at 12 59 04 pm](https://user-images.githubusercontent.com/16470658/30708465-fe8c75b4-9ecc-11e7-9be7-d8d010880db1.png)

- Namespaces colored differently depending on whether they're in front of a class name or not
- Class names being yellow when they're an import, a class definition or an instantiation
- Class names being green when they're a parent class
- Class names being purple when used as types in function parameters

It adds a lot of visual noise. I know these can be overridden by the stylesheet, but it presents an odd first experience.

**Comment by dsifford** [comment]
Re: Class tokenization for type hinting. IMHO, it's currently scoped correctly. Classes being defined or instantiated are fundamentally different (and should be scoped as such) than those being used for type hinting. This is consistent with all other languages to my knowledge.

@philraj To comment specifically on the issues you're having: These are all exclusively issues with the theme you're using. 

I am the maintainer of the Dracula theme for VSCode (which uses the same grammars as Atom) and here's how your example looks in my editor right now:

![image](https://user-images.githubusercontent.com/5240018/31208415-ae0533ca-a952-11e7-997d-c82682807802.png)

To summarize: I think your concerns are totally valid. The color inconsistencies in your example are _totally_ distracting. However, I think this issue should be handled by the theme maintainers, rather than the grammar.

**Comment by philraj** [comment]
@dsifford Yes that looks much better! I'll take it up with the One Dark folks; that's sort of what I figured after giving it some thought. Thanks.

**Comment by 50Wliu** [comment]
:+1: I think it's safe to close this issue then.
